tests/ui/proc-macro/issue-89699.rs

```diff
@@ -0,0 +1,6 @@
+// aux-build:async-move.rs
```

Comment:
And here as well, let's add a comment like "Non-regression test for issue 89699 where a proc-macro emitting edition 2018 syntax is used on edition 2015"

Even if it's the default, it's kind of the point of the test so maybe we could also make the `// edition: 2015` explicit (though I'm not sure if the annotation works with the default value). I'll let you ponder this question, I'm fine either way.